### PR TITLE
Week2

### PR DESCRIPTION
**Pull request by cremerf**
## Release Notes

- **New Features**
  - Introduced data preprocessing and machine learning model training workflows with MLflow integration.
  - Added new scripts for logging and registering models, including support for custom models and feature engineering.
  - Implemented a `CancellationModel` class for model evaluation and prediction.
  - Added a new `Preprocessor` class for data cleaning and transformation.

- **Bug Fixes**
  - Updated configuration parameters for improved model performance.

- **Documentation**
  - Added JSON files to define MLflow experiments and model versions for better tracking.

- **Chores**
  - Updated project metadata and dependencies in configuration files.
  - Removed obsolete files related to path management and preprocessing.

### CODE REVIEW

File: src/hotel_reservation/paths.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class AllPaths:
+    def __init__(self) -> None:
+        load_dotenv()  # Load environment variables from .env file if present
+
+        if self.is_databricks():
+            # Running in Databricks
+            self.filename_config = Path(
+                os.environ.get(
+                    "DATABRICKS_CONFIG_PATH",
+                    "/Workspace/Users/cremerfederico29@gmail.com/marvelmlops-cremerf/project-config.yml",
```
Comment (mvechtomova) [inline src/hotel_reservation/paths.py:17]: This should be dynamic. You will see next week, keep it for now

File: src/hotel_reservation/paths.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                )
+            )
+        else:
+            # Running in local environment
+            self.BASE_DIR = Path(__file__).resolve().parent.parent.parent
+            self.filename_config = self.BASE_DIR / "project-config.yml"
+
+        self.config = self.get_config_file()
+        self.cremerf_catalog = f'/Volumes/{self.config["catalog_name"]}/{self.config["schema_name"]}/'
```
Comment (mvechtomova) [inline src/hotel_reservation/paths.py:26]: I would avoid adding your name into attribute names

File: src/hotel_reservation/paths.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        self.data_volume = f"{self.cremerf_catalog}data/"
+
+    def is_databricks(self):
+        is_db_env = os.environ.get("IS_DATABRICKS")
```
Comment (mvechtomova) [inline src/hotel_reservation/paths.py:30]: not sure why that is useful to check where you run

**Review by mvechtomova — APPROVED** [review]
added some comments, will approve for now :-)

File: src/hotel_reservation/paths.py (hunk 1 above)
Comment (cremerf) [reply src/hotel_reservation/paths.py:17]: took note to remind it later.

File: src/hotel_reservation/paths.py (hunk 2 above)
Comment (cremerf) [reply src/hotel_reservation/paths.py:26]: i'll change in the next submission

**Pull request merged**